### DISCUSSION TITLE
What does "(*)" mean in the commits log?

### DISCUSSION THREAD

**Discussion by jbrains** [jesseduffield/lazygit#2568] [Q&A] [answered]
Hi, folks. I just noticed "(\*)" on some of my commits in the Commits log. I tried to search for an answer, but I guess "(\*)" is a difficult string to search for. :) What does this asterisk signify about a commit?

**Comment by stefanhaller** [comment, marked as answer]
This is very new on master. It means that there is some other branch head pointing to this commit (including your upstream, if you have one). If you press `+` to expand the commits panel, you can see the full names of these branches; the asterisk is meant to be an abbreviation of that.

It's one step on a longer journey to support stacked branches better. With stacked branches it's important to see where one branch ends and the next one starts. This doesn't work very well on master yet, but will soon; see #2552 (and #2527).

**Reply by peppy** [reply]
Thanks for the explanation. I feel that this is very hard to understand in its current implementation, and leads to more confusion than anything else (initially I thought this was part of the commit message, and had to actually go into the lazygit source to figure out where it came from).

For reference:

https://github.com/jesseduffield/lazygit/blob/c6c4346d4886dbfcfe3a164170f99936734febd8/pkg/gui/presentation/commits.go#L273-L283

**Reply by jesseduffield** [reply]
I also find it a bit onerous: I don't use a stacked branches workflow myself, but I have found that the branch markers create quite a bit of clutter without giving me much information. Perhaps it's better to leave the extra detail to the expanded view?

Or perhaps there should be a way in lazygit for a user to say branch X is based on branch Y and so we can show a marker in the non-expanded view signifying that (but ignoring everything else). That way there won't be as much clutter.

If people agree, I'm happy to scrap the marker in the non-expanded view for now until we come up with a better UX. Thoughts @stefanhaller @peppy @jbrains ?

**Reply by stk-ableton** [reply]
Showing it only in the expanded view is definitely not good enough for me. I work with stacked branches a lot, and I find it essential to see the individual branch heads in the non-expanded view. That said, I do find it a little bit unfortunate that the current `(*)` UI doesn't let me distinguish between another branch and my upstream, which is visualized in the same way.

Having to tell lazygit how my branches are stacked doesn't seem right to me; git knows this, I shouldn't have to configure this in lazygit.

If we can't find a way to visualize the information in a less confusing way, we may have to make it a config (opt-in or opt-out?). One idea might be to make it dependent on the `rebase.updateRefs` git config, because people who turn that on probably care about seeing their branch heads.

**Reply by peppy** [reply]
For me it is not helpful to my workflow. It's great to have it displayed in the expanded view, but personally I don't benefit from having the information visible in the compact view.

For instance, this is me committing a new change on top of `master`:

![iTerm 2023-05-02 at 06 48 14](https://user-images.githubusercontent.com/191335/235598531-7706327f-faed-40c0-9073-ffe017a260b6.png)

Expanded:

![iTerm 2023-05-02 at 06 48 42](https://user-images.githubusercontent.com/191335/235598578-d6cf985f-c586-4d99-8c42-0837b238d10a.png)

A lot of the branch references are to branches I have recently checked out to review a pull request. This gets noisy quickly.

**Reply by jesseduffield** [reply]
I'm thinking we make it an explicit opt-in for now and then think up a way to improve the UX. We may land on an approach that is both informative and compact such that we feel comfortable enabling it by default, but it's currently uncharted territory and needs some experimentation

**Reply by jesseduffield** [reply]
I should mention that we already by default only show the branch visualisation in the expanded view and I've opted in to also showing it in the compact view but it's not something I'd want to enable by default given how much space that visualisation takes up.

**Reply by stefanhaller** [reply]
OK, happy to make a PR if you tell me what config name to use for this, and which section it should go in.

**Reply by jesseduffield** [reply]
I'm thinking `gui.experimentalShowBranchHeads` (boolean). `experimental` because we may land on a solution that's less about showing all branch heads and more about showing relevant branch heads, which would lead to a more appropriate name.

**Reply by stefanhaller** [reply]
https://github.com/jesseduffield/lazygit/pull/2582

**Reply by jesseduffield** [reply]
This is now addressed in https://github.com/jesseduffield/lazygit/releases/tag/v0.38.1. Thanks @stefanhaller for the PR
